### PR TITLE
Feat: Add model to promote

### PR DESCRIPTION
**Pull request by eakmanrq**

### CODE REVIEW

File: sqlmesh/schedulers/airflow/plan.py [hunk 1]
```diff
@@ -82,7 +82,9 @@ def create_plan_dag_spec(
         environment_naming_info=request.environment.naming_info,
         new_snapshots=request.new_snapshots,
         backfill_intervals_per_snapshot=backfill_intervals_per_snapshot,
-        promoted_snapshots=request.environment.snapshots,
+        promoted_snapshots=[
+            x for x in all_snapshots.values() if x.table_info in request.environment.snapshots
```
Comment (izeigerman) [inline sqlmesh/schedulers/airflow/plan.py:86]: Better iterate through `request.environment.snapshots` and then access the name in `all_snapshots` with O(1). Right now this expression is O(n^2(

Comment (izeigerman) [reply sqlmesh/schedulers/airflow/plan.py:86]: Btw, storing all these snapshots in the spec seems rather excessive. Can't we get them in the `dag_generator` instead while creating a promotion task?

Comment (eakmanrq) [reply sqlmesh/schedulers/airflow/plan.py]: Thanks great points. Check out latest version and let me know what you think.

Comment (izeigerman) [reply sqlmesh/schedulers/airflow/plan.py]: Looks much better, thanks! Is is this still a draft? I think this is is good to go otherwise.

Comment (eakmanrq) [reply sqlmesh/schedulers/airflow/plan.py]: I was hoping to verify it works at the client before merging.

**Pull request merged**